main.go

```diff
@@ -111,6 +116,16 @@ func main() {
 	}
 
 	if socket != "" {
+		go func() {
+			for {
+				if ! isParentAlive() {
+					os.Exit(0)
```

Comment:
It will be better to add some logging when this happens, so we know why the server quits.